### DISCUSSION TITLE
How to build for armv7-linux with custom python distribution

### DISCUSSION THREAD

**Discussion by 124C41p** [indygreg/PyOxidizer#696] [Q&A]
I understand that armv7 is not officially supported yet, as it was already mentioned [here](https://github.com/indygreg/PyOxidizer/issues/361). But can I make PyOxidizer to build for that target architecture by providing a custom Python distribution?

What I have done so far:

1. Create a custom python distribution
```
git clone https://github.com/indygreg/python-build-standalone.git
cd python-build-standalone
./build-linux.py --target armv7-unknown-linux-gnueabihf
```

2. Creating an armv7 docker image based on debian jessie with rust and pyoxidizer (version 0.24.0, installed by cargo install) available

3. Create a new pyoxidizer project inside a corresponding armv7 docker container
```
pyoxidizer init-rust-project --system-rust foo
cd foo
```
Inside `pyoxidizer.bzl` I replaced `dist = default_python_distribution()` by `dist = PythonDistribution(local_path="...", sha256="...")`.
```
pyoxidizer build --system-rust --target-triple armv7-unknown-linux-gnueabihf
```

(The `target-triple` argument is necessary, since otherwise PyOxidizer would try to call `x86_64-linux-gnu-gcc`, which is of course not available on my armv7 host system.)

Unfortunately, the build process stops due to the following error:
```
   Compiling pyo3-ffi v0.17.3
error: failed to run custom build command for `pyo3-ffi v0.17.3`

Caused by:
  process didn't exit successfully: `/tmp/pyoxidizerVBGVrl/build/target/debug/build/pyo3-ffi-5f4a9940165a7bbf/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=PYO3_PRINT_CONFIG

  --- stderr
  error: your Rust target architecture (32-bit) does not match your python interpreter (64-bit)
```

This is really strange since I tested my custom python build on a Raspberry Pi 4. So I am pretty sure that my python interpreter is not a 64-bit application.

Any ideas?

**Comment by inklesspen** [comment]
It looks like PyO3 0.9.2 [fixed a bug](https://github.com/PyO3/pyo3/pull/779) where 32-bit Python was not properly detected, but since PyOxidizer hasn't been updated in a while, it doesn't depend on a version with that bugfix.
